components/GameScreen.vue

```diff
@@ -179,87 +263,44 @@ watch(gameState, async (newState, prevState) => {
       autoDensity: true,
     });
 
-    let isZoomingOut = false;
-    const zoomDuration = 500; // Duration of the zoom-out effect in milliseconds
-    let startZoomTime: number | null = null;
-    let startMousePos = { x: 0, y: 0 };
-
-    // Functions to handle smooth zoom out
-    function smoothZoomOut(mousePos: { x: number; y: number }) {
-      if (!appRef.value) {
-        return;
+    // Follow player bot
+    function follow() {
```

Comment:
Can we move code of this function to the `watch(isFollowing, () => {});` callback?

This way the button's `@click` can be `@click="isFollowing = !isFollowing"` and mode toggle will be a reaction to state change?

Reply:
I dont think I can move this function outside of ```watch(gameState, async (newState, prevState) => {```.
I need to use ```const app = new Application();``` inside watch or it will cause this error:
![image](https://github.com/user-attachments/assets/2fbc625e-218e-4da3-b6db-557f5ac417b3)

I did try moving it outside and use ```appRef.value``` instead of ``` const app``` it causes that error on first load.

Reply:
@aldrin312, why does the error happen? `appRef` initializes on the first fetch before the screen is getting rendered.

And the user can't zoom until the game screen is rendered anyway 🤔

Reply:
I'm not sure but it is the ```appRef``` that is causing it when I moved it outside.
If it inside ```watch(gameState, async (newState, prevState)``` it works fine.

Reply:
@aldrin312, will calling `appRef.value?.stage...` help?

Maybe, we can check if `appRef.value` is initialized. Should help with possible races.

Reply:
Could possibly a bug with Vue maybe?

Reply:
@aldrin312, can you please check if using `shallowRef` for `appRef` instead of `ref` will help?

Reply:
Something is happening inside ```watch(gameState, async (newState, prevState)``` that's initializing what it needs.
Since when I zoom in with mouse wheel first, everything works fine after.

Reply:
@aldrin312, weird. We may look into it separately.